src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -679,25 +686,8 @@ public BibEntry withField(String field, String value) {
     * Returns user comments (arbitrary text before the entry), if they exist. If not, returns the empty String
      */
     public String getUserComments() {
-
-        if (parsedSerialization != null) {
-
-            try {
-                // get the text before the entry
-                String prolog = parsedSerialization.substring(0, parsedSerialization.lastIndexOf('@'));
-
-                // delete trailing whitespaces (between entry and text)
-                prolog = prolog.replaceFirst("\\s+$", "");
-
-                // if there is any non whitespace text, write it
-                if (prolog.length() > 0) {
-                    return prolog;
-                }
-            } catch (StringIndexOutOfBoundsException ignore) {
-                // if this occurs a broken parsed serialization has been set, so just do nothing
-            }
-        }
-        return "";
+        // delete trailing whitespaces (between entry and text) from stored serialization
+        return commentsBeforeEntry.replaceFirst("\\s+$", "");
```

Comment:
Okay. Then I would suggest extracting the Pattern with Pattern.compile and use Matcher.replaceF'irst...
Discussion about it on SO: http://stackoverflow.com/questions/1466959/string-replaceall-vs-matcher-replaceall-performance-differences
https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#replaceFirst(java.lang.String)